gammapy/background/models.py

```diff
@@ -71,3 +80,820 @@ def evaluate(self, x, y):
+        detx_bins = Angle(detx_bins, detx_unit)
+        dety_bins = Angle(dety_bins, dety_unit)
+
+        # get energy binning
+        energy_bins = _make_bin_edges_array(data['ENERG_LO'], data['ENERG_HI'])
+        if header['TUNIT5'] == header['TUNIT6']:
+            energy_unit = header['TUNIT5']
+        else:
+            raise ValueError("Energy units not matching ({0}, {1})"
+                             .format(header['TUNIT5'], header['TUNIT6']))
+        energy_bins = Quantity(energy_bins, energy_unit)
+
+        # get background data
+        background = data['Bgd'][0]
+        background_unit = header['TUNIT7']
+        TeV_units = ['1/s/TeV/sr', 's-1 sr-1 TeV-1', '1 / (s sr TeV)',
+                     '1 / (TeV s sr)']
+        MeV_units = ['1/s/MeV/sr', 'MeV-1 s-1 sr-1', '1 / (s sr MeV)',
+                     '1 / (MeV s sr)']
+        if background_unit in TeV_units:
+            background_unit = '1 / (s TeV sr)'
+        elif background_unit in MeV_units:
+            background_unit = '1 / (s MeV sr)'
+        else:
+            raise ValueError("Cannot interpret units ({})".format(background_unit))
+        background = Quantity(background, background_unit)
+
+        return CubeBackgroundModel(detx_bins=detx_bins,
+                                   dety_bins=dety_bins,
+                                   energy_bins=energy_bins,
+                                   background=background)
+
+    @staticmethod
+    def from_fits_image(tbhdu):
+        """Read cube background model from image in fits file.
+
+        Parameters
+        ----------
+        tbhdu : `~astropy.io.fits.BinTableHDU`
+            HDU binary table for the bg cube
+
+        Returns
+        -------
+        bg_cube : `~gammapy.models.CubeBackgroundModel`
+            bg model cube object
+        """
+        raise NotImplementedError
+
+
+    @staticmethod
+    def read_bin_table(filename):
+        """Read cube background model from binary table in fits file.
+
+        Parameters
+        ----------
+        filename : `~string`
+            name of file with the bg cube
+
+        Returns
+        -------
+        bg_cube : `~gammapy.models.CubeBackgroundModel`
+            bg model cube object
+        """
+        hdu = fits.open(filename)
+        return CubeBackgroundModel.from_fits_bin_table(hdu)
+
+
+    @staticmethod
+    def read_image(filename):
+        """Read cube background model from image in fits file.
+
+        Parameters
+        ----------
+        filename : `~string`
+            name of file with the bg cube
+
+        Returns
+        -------
+        bg_cube : `~gammapy.models.CubeBackgroundModel`
+            bg model cube object
+        """
+        hdu = fits.open(filename)
+        return CubeBackgroundModel.from_fits_image(hdu)        
+
+
+    def to_astropy_table(self):
+        """Convert cube background model to astropy table format.
+
+        Returns
+        -------
+        name : `~string`
```

Comment:
Yes, please do it.

If I want an Astropy table I want the name in `meta` ... that's what `meta` is for, a place to store some meta info instead of handing it around separately.

I can still put the name as the `BinTableHDU` FITS extension name when I want to store as FITS.
You can do this in the FITS writer, no?